### ISSUE TITLE
Rule with Lessert TimePeriod format always overrides all other rules.

### ISSUE THREAD

**Issue by amarentheran** [stefanprodan/AspNetCoreRateLimit#86]
Hello,

I have implemented ClientRateLimiting where in GeneralRules am setting to Period:1s, Limit:11
and in ClientRules am setting the Period:1m, Limit: 3. 
But always the GeneralRules gets overriden.
Also I tried same config by interchanging the rules and I got exactly opposite Results.

`  "ClientRateLimiting": {
    "EnableEndpointRateLimiting": true,
    "ClientIdHeader": "Ocp-Apim-Subscription-Key",
    "HttpStatusCode": 429,
    "ClientWhitelist": [ "cl-key-a", "cl-key-b" ],
    "GeneralRules": [
      {
        "Endpoint": "*",
        "Period": "1m",
        "Limit": 11
      }
    ]
  },
  "ClientRateLimitPolicies": {
    "ClientRules": [
      {
        "ClientId": "070f1593-3dd8-48d4-88f1-531182e204a8",
        "Rules": [
          {
            "Endpoint": "*",
            "Period": "1s",
            "Limit": 3
          }
        ]
      }
    ]
  }`

ResponseMessage on getting throttled, Here GeneralRules got picked up instead of ClientRule.
`"API calls quota exceeded! maximum admitted 11 per 1m."
`
**But when I gave same TimePeriod Format for both Rules, it gets executed properly.**
ClientWhitelist also works fine.

Here's the API call snippet:
```
using (HttpClient httpClient = new HttpClient())
{
httpClient.DefaultRequestHeaders.Accept.Clear();
httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
httpClient.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", "070f1593-3dd8-48d4-88f1-531182e204a8");
string requestUri = "https://localhost:44337/{apiMethodName}";
var httpResponseMessage = httpClient.GetAsync(requestUri).Result;
}

```
Can you please look into the issue why giving different TimePeriodFormats is causing the issue,

**Comment by randaarchitect** [comment]
What happens if you add a "1m" rule to the client, like this:

{ "Endpoint": "*", "Period": "1m", "Limit": 180 }

**Comment by amarentheran** [comment]
It works as both have same time period.

**Comment by heartFireTree** [comment]
No, the real rule is that a smaller quota limit will cover a larger quota
